components/nvs_flash/src/nvs_page.cpp

```diff
@@ -308,6 +308,63 @@ esp_err_t Page::readItem(uint8_t nsIndex, ItemType datatype, const char* key, vo
     return ESP_OK;
 }
 
+esp_err_t Page::cmpItem(uint8_t nsIndex, ItemType datatype, const char* key, const void* data, size_t dataSize, uint8_t chunkIdx, VerOffset chunkStart)
+{
+    size_t index = 0;
+    Item item;
+
+    if (mState == PageState::INVALID) {
+        return ESP_ERR_NVS_INVALID_STATE;
+    }
+
+    esp_err_t rc = findItem(nsIndex, datatype, key, index, item, chunkIdx, chunkStart);
+    if (rc != ESP_OK) {
+        return rc;
+    }
+
+    if (!isVariableLengthType(datatype)) {
+        if (dataSize != getAlignmentForType(datatype)) {
+            return ESP_ERR_NVS_TYPE_MISMATCH;
+        }
+
+        if(!memcmp(data, item.data, dataSize))
+        {
+            return ESP_ERR_NVS_COMPARE_IDENTICAL;
+        }
+        return ESP_OK;
+    }
+
+    if (dataSize < static_cast<size_t>(item.varLength.dataSize)) {
+        return ESP_ERR_NVS_INVALID_LENGTH;
+    }
+
+    bool isSame = true;
+    const uint8_t* dst = reinterpret_cast<const uint8_t*>(data);
+    size_t left = item.varLength.dataSize;
+    for (size_t i = index + 1; i < index + item.span; ++i) {
```

Comment:
Since we only care if this function returns `ESP_ERR_NVS_COMPARE_IDENTICAL` or not, is it possible to exit this loop early if `!isSame`, and skip the CRC check?

In this case, it might be nicer to redefine the return codes of this function: `ESP_OK` — content is the same; `ESP_ERR_NVS_CONTENT_DIFFERS` or `ESP_ERR_NOT_FOUND` — other possible return values.